python/swiftnav/nav_msg.pyx

```diff
@@ -16,8 +16,8 @@ from signal cimport gnss_signal_t
 
 cdef class NavMsg:
 
-  def __cinit__(self):
-    nav_msg_init(&self._thisptr)
+  def __cinit__(self, sid):
```

Comment:
err, yes. so `_thisptr` actually refers to the underlying C struct, it's not actually a pointer, so `sid._thisptr` actually passes it by value.